questionpy_server/models.py

```diff
@@ -91,13 +91,23 @@ class AttemptScoreArguments(AttemptViewArguments):
     generate_hint: bool
 
 
-class NotFoundStatusWhat(Enum):
-    PACKAGE = "PACKAGE"
-    QUESTION_STATE = "QUESTION_STATE"
-
+class RequestErrorCode(Enum):
+    QUEUE_WAITING_TIMEOUT = "QUEUE_WAITING_TIMEOUT"
+    WORKER_TIMEOUT = "WORKER_TIMEOUT"
+    OUT_OF_MEMORY = "OUT_OF_MEMORY"
+    INVALID_PACKAGE = "INVALID_PACKAGE"
+    INVALID_REQUEST = "INVALID_REQUEST"
+    PACKAGE_ERROR = "PACKAGE_ERROR"
+    CALLBACK_API_ERROR = "CALLBACK_API_ERROR"
+    SERVER_ERROR = "SERVER_ERROR"
```

Comment:
1. Ich würde mir mehr Trennschärfe wünschen. Zumindest folgende Fälle halte ich es für sinnvoll, zu unterscheiden, damit wir passende Fehlermeldungen anzeigen können:
   - Paket existiert nicht
   - Options invalide
   - Question state kaputt
   - Attempt state kaputt
3. Hier ginge auch 
    ```suggestion
    class RequestErrorCode(StrEnum):
        QUEUE_WAITING_TIMEOUT = auto()
        WORKER_TIMEOUT = auto()
        OUT_OF_MEMORY = auto()
        INVALID_PACKAGE = auto()
        INVALID_REQUEST = auto()
        PACKAGE_ERROR = auto()
        CALLBACK_API_ERROR = auto()
        SERVER_ERROR = auto()
    ```

    für weniger duplication.

Reply:
1. Wenn wir in allen Fehlern (status code != 2xx) ein gleichartiges JSON-Objekt zurückgeben, was ich begrüße, ist der Fall "Paket existiert nicht" natürlich sinnvoll. Die anderen drei genannten Fälle fallen eigentlich unter den generischen `PACKAGE_ERROR`. Wir können hier aber auch differenzieren. `InvalidQuestionStateError` ist zumindest schon definiert in common, wird benutzt und kann dann leicht als Fehlerfall kommuniziert werden.
Welchen Fehlerfall meinst du bei "Options invalide"?

3. "Using [auto](https://docs.python.org/3/library/enum.html#enum.auto) with [StrEnum](https://docs.python.org/3/library/enum.html#enum.StrEnum) results in the lower-cased member name as the value."
ist daher in diesem Fall nicht so gut 
Generell würde ich `auto()` nicht im Zusammenhang mit Schnittstellen benutzen, da sich das Verhalten möglicherweise in neueren Versionen ändern kann.

Reply:
Mit "Options invalide" meine ich `OptionsFormValidationError`, wobei das tbf auch auf #126 warten musste.

Ich denke, "Question state kaputt" und "Attempt state kaputt" wäre insofern hilfreich zu unterscheiden, dass wir dann in der Fehlermeldung erkennbar machen können, dass entweder die ganze Frage oder "nur" der Attempt fehlerhaft ist. Das hilft zwar keiner Studi, aber womöglich schon Lehrenden (und Entwickler:innen eh)

Reply:
Bezüglich `OptionsFormValidationError` kommen wir an einen guten Punkt, den wir aber nicht hier angehen sollten. Ich habe dafür Issue #122 erstellt.

> Ich denke, "Question state kaputt" und "Attempt state kaputt" wäre insofern hilfreich zu unterscheiden, dass wir dann in der Fehlermeldung erkennbar machen können, dass entweder die ganze Frage oder "nur" der Attempt fehlerhaft ist. Das hilft zwar keiner Studi, aber womöglich schon Lehrenden (und Entwickler:innen eh)

Dem schließe ich mich an. @janbritz Bitte führe in common noch eine neue Exception `InvalidAttemptStateError` ein und ergänze zwei neue Error Codes `INVALID_QUESTION_STATE` und `INVALID_ATTEMPT_STATE`.